test/test_lagrangian_particle_tracking.jl

```diff
@@ -269,6 +269,19 @@ lagrangian_particle_test_grid(arch, ::Periodic, z) =
 lagrangian_particle_test_grid(arch, ::Flat, z) =
     RectilinearGrid(arch; topology=(Periodic, Flat, Bounded), size=(5, 5), x=(-1, 1), z)
 
+lagrangian_particle_test_grid_expanded(arch, ::Periodic, z) =
+    RectilinearGrid(arch; topology=(Periodic, Periodic, Bounded), size=(5, 5, 5), x=(-1, 1), y=(-1, 1), z = 2 .*z)
+lagrangian_particle_test_grid_expanded(arch, ::Flat, z) =
+    RectilinearGrid(arch; topology=(Periodic, Flat, Bounded), size=(5, 5), x=(-1, 1), z = 2 .*z)
```

Comment:
I'm trying to add this but bumping against this error:

```julia
julia> grid = LatitudeLongitudeGrid(topology=(Bounded, Flat, Bounded), size = (5, 5), longitude=(-1, 1), z=(-1, 0))
5×1×5 LatitudeLongitudeGrid{Float64, Bounded, Flat, Bounded} on CPU with 3×0×3 halo and without precomputed metrics
├── longitude: Bounded  λ ∈ [-1.0, 1.0] regularly spaced with Δλ=0.4
├── latitude:  Flat φ                   
└── z:         Bounded  z ∈ [-1.0, 0.0] regularly spaced with Δz=0.2

julia> NonhydrostaticModel(; grid)
ERROR: None of the implemented pressure solvers for NonhydrostaticModel currently support more than one stretched direction.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] PressureSolver(arch::CPU, grid::LatitudeLongitudeGrid{Float64, Bounded, Flat, Bounded, Nothing, Nothing, Float64, Float64, Float64, OffsetVector{…}, Nothing, OffsetVector{…}, CPU})
   @ Oceananigans.Models.NonhydrostaticModels ~/repos/Oceananigans.jl2/src/Models/NonhydrostaticModels/NonhydrostaticModels.jl:61
 [3] NonhydrostaticModel(; grid::LatitudeLongitudeGrid{…}, clock::Clock{…}, advection::Centered{…}, buoyancy::Nothing, coriolis::Nothing, stokes_drift::Nothing, forcing::@NamedTuple{}, closure::Nothing, boundary_conditions::@NamedTuple{}, tracers::Tuple{}, timestepper::Symbol, background_fields::@NamedTuple{}, particles::Nothing, biogeochemistry::Nothing, velocities::Nothing, hydrostatic_pressure_anomaly::Oceananigans.Models.NonhydrostaticModels.DefaultHydrostaticPressureAnomaly, nonhydrostatic_pressure::Field{…}, diffusivity_fields::Nothing, pressure_solver::Nothing, immersed_boundary::Nothing, auxiliary_fields::@NamedTuple{})
```

Looking at the methods for `PressureSolver()`, I understand why this error is being triggered. But the pressure should actually be able to deal with this case, no?

Just trying to avoid setting up a separate test using the hydrostatic model.

Reply:
Ok, that makes sense; we do not have a pressure solver for curvilinear grids yet. Once the one for immersed boundaries is fully functional, it will work also on curvilinear grids (albeit with no preconditioner). Why don't you use the hydrostatic model with prescribed velocities for the test? It is quite simpler and faster because of this:

```julia

# If the model only tracks particles... do nothing but that!!!
const OnlyParticleTrackingModel = HydrostaticFreeSurfaceModel{TS, E, A, S, G, T, V, B, R, F, P, U, C} where
                 {TS, E, A, S, G, T, V, B, R, F, P<:AbstractLagrangianParticles, U<:PrescribedVelocityFields, C<:NamedTuple{(), Tuple{}}}

function time_step!(model::OnlyParticleTrackingModel, Δt; callbacks = [], kwargs...)
    tick!(model.clock, Δt)
    model.clock.last_Δt = Δt
    step_lagrangian_particles!(model, Δt)
    update_state!(model, callbacks)
end

update_state!(model::OnlyParticleTrackingModel, callbacks) =
    [callback(model) for callback in callbacks if callback.callsite isa UpdateStateCallsite]
```

Reply:
I wasn't aware of that, interesting! I'll try it out :)

Reply:
Added LatLon tests with HydrostaticModel!

Reply:
nice